faiss/gpu/impl/IVFPQ.cu

```diff
@@ -256,6 +256,35 @@ void IVFPQ::appendVectors_(
     }
 }
 
+void IVFPQ::removeVectors_(
+        Tensor<int, 1, true>& listIds,
+        Tensor<int, 1, true>& listOffset,
+        Tensor<int, 1, true>& listReplaceOffset,
+        Tensor<long, 1, true>& listIndicesReplaceOffset,
+        cudaStream_t stream) {
+    runIVFIndicesRemove(
+            listIds,
+            listOffset,
+            listReplaceOffset,
+            listIndicesReplaceOffset,
+            indicesOptions_,
+            deviceListIndexPointers_,
+            stream);
+
+    if (interleavedLayout_) {
```

Comment:
A substantial amount of work has already been done up until this point (e.g., `IVFBase::removeVectors`). This check should be the first thing that happens if you call remove, and change the assert to a throw instead